### PR TITLE
sqlitestudio: init at 3.4.4

### PR DESCRIPTION
**Pull request by AsterisMono**
## Description of changes

Adds [SQLiteStudio](https://sqlitestudio.pl/), a free, open source, multi-platform SQLite database manager.
License: GPLv3

[Build instructions](https://github.com/pawelsalawa/sqlitestudio/wiki/Instructions_for_compilation_under_Linux)
[AUR PKGBUILD](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=sqlitestudio)

I'm a nix novice and this is my first-time contribution to Nixpkgs. I've read the contribution guide and tried my best to follow, please let me know if something is wrong or needs to be changed. Thank you!

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sq/sqlitestudio/package.nix [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    [
+      qtbase
+      qtsvg
+    ]
+    ++ [
+      readline
+      tcl
+    ];
+
+  qmakeFlags = [ "../../SQLiteStudio3" ];
```
Comment (AsterisMono) [inline pkgs/by-name/sq/sqlitestudio/package.nix:42]: According to build instructions [here](https://github.com/pawelsalawa/sqlitestudio/wiki/Instructions_for_compilation_under_Linux#main-project) we need to run `qmake ../../SQLiteStudio3`.

**Comment by AsterisMono** [comment]
SQLiteStudio also provides [official plugins](https://github.com/pawelsalawa/sqlitestudio/wiki/Instructions_for_compilation_under_Linux#official-plugins). Should I package them separately (in another package or in another PR), or include them in the base package?

File: pkgs/by-name/sq/sqlitestudio/package.nix [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      $out/share/pixmaps/sqlitestudio.svg
+  '';
+
+  enableParallelBuilding = true;
+
+  meta = with lib; {
+    description = "A free, open source, multi-platform SQLite database manager";
+    homepage = "https://sqlitestudio.pl/";
+    license = licenses.gpl3;
+    mainProgram = pname;
```
Comment (kirillrdy) [inline pkgs/by-name/sq/sqlitestudio/package.nix:78]: usage of pname is considered anti-pattern https://github.com/NixOS/nixpkgs/issues/277994

File: pkgs/by-name/sq/sqlitestudio/package.nix [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      readline
+      tcl
+      python3
+    ];
+
+  qmakeFlags = [ "../../SQLiteStudio3" ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = pname;
```
Comment (kirillrdy) [inline pkgs/by-name/sq/sqlitestudio/package.nix:50]: same here

File: pkgs/by-name/sq/sqlitestudio/package.nix [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+  src = fetchFromGitHub {
+    owner = "pawelsalawa";
+    repo = "sqlitestudio";
+    rev = version;
+    hash = "sha256-5oBYv8WxyfVvvqr15XApvn6P/lBxR8b6E+2acRkvX0U=";
+  };
+
+  nativeBuildInputs =
+    with libsForQt5.qt5;
```
Comment (Aleksanaa) [inline pkgs/by-name/sq/sqlitestudio/package.nix:24]: If you want to let `copyDesktopItems` escape `with`, it should be put before `with`.

File: pkgs/by-name/sq/sqlitestudio/package.nix [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  postInstall = ''
+    install -Dm755 \
+      ../../SQLiteStudio3/guiSQLiteStudio/img/sqlitestudio.svg \
+      $out/share/pixmaps/sqlitestudio.svg
+  '';
+
+  enableParallelBuilding = true;
+
+  meta = with lib; {
+    description = "A free, open source, multi-platform SQLite database manager";
```
Comment (Aleksanaa) [inline pkgs/by-name/sq/sqlitestudio/package.nix:75]:
```suggestion
    description = "Free, open source, multi-platform SQLite database manager";
```

File: pkgs/by-name/sq/sqlitestudio/package.nix [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+  postInstall = ''
+    install -Dm755 \
+      ../../SQLiteStudio3/guiSQLiteStudio/img/sqlitestudio.svg \
+      $out/share/pixmaps/sqlitestudio.svg
+  '';
+
+  enableParallelBuilding = true;
+
+  meta = with lib; {
```
Comment (Aleksanaa) [inline pkgs/by-name/sq/sqlitestudio/package.nix:74]:
```suggestion
  meta = {
```

File: pkgs/by-name/sq/sqlitestudio/package.nix [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+      terminal = false;
+      startupNotify = false;
+      categories = [ "Development" ];
+    })
+  ];
+
+  preConfigure = ''
+    mkdir -p output/build
+    cd output/build
+  '';
```
Comment (Aleksanaa) [inline pkgs/by-name/sq/sqlitestudio/package.nix:64]: This should not be necessary?

File: pkgs/by-name/sq/sqlitestudio/package.nix (hunk 2 above)
Comment (AsterisMono) [reply pkgs/by-name/sq/sqlitestudio/package.nix]: Thank you! I have removed all pname references.

File: pkgs/by-name/sq/sqlitestudio/package.nix (hunk 7 above)
Comment (AsterisMono) [reply pkgs/by-name/sq/sqlitestudio/package.nix]: Yes indeed, this is not necessary and has been removed.
`qmakeFlags` and `postInstall` instructions has been modified accordingly. Thank you!

**Comment by Aleksanaa** [comment]
@ofborg build sqlitestudio

**Pull request merged**